src/components/ui/search-anonymized.tsx

```diff
@@ -0,0 +1,118 @@
+"use client";
+
+import type { Section } from "~/models/types";
+import { Form, FormControl, FormField, FormItem } from "./form";
+import { z } from "zod";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+import { Label } from "./label";
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "./select";
+import { useEffect } from "react";
+import type { BusinessUnit } from "@prisma/client";
+
+
+const formSchema = z.object({
+    role: z.string(),
+    unit: z.string()
+})
+
+export default function SearchAnonymized({roles, businessUnits} : {roles: Section[], businessUnits : BusinessUnit[]}) {
+    const searchParams = useSearchParams();
+    const path = usePathname();
+    const route = useRouter();
+    let previousRole = "";
+    let previousUnit = "";
+
+    const form = useForm<z.infer<typeof formSchema>>({
+        resolver: zodResolver(formSchema),
+        defaultValues: {
+          role: searchParams.get("role") ?? "",
+          unit: searchParams.get("unit") ?? "",
+        },
+    });
+
+    function onSubmit(values: z.infer<typeof formSchema>) {
+        const role2 = values.role != undefined ? values.role.length != 0 && values.role != "No role" ? `role=${values.role}` : "" : "";
+        const unit2 = values.unit != undefined ? values.unit.length != 0 && values.unit != "No unit" ? `&unit=${values.unit}` : "" : "";
+        route.push(`${path}?${role2}${unit2}`);
+    }
+
+    useEffect(() => {
+        setInterval(() => {
+            const currentRole = form.getValues().role;
+            const currentUnit = form.getValues().unit;
+            if (previousRole != currentRole || previousUnit != currentUnit) {
+                previousRole = currentRole;
+                previousUnit = currentUnit;
+                onSubmit({role: currentRole, unit: currentUnit});
+            }
+            
+        }, 1000);
```

Comment:
Was your intention here to use this `setInterval` as a debounce? If so, then it's maybe better to have a look at `setTimeout` instead. It will fire unintentionally when no actual update has happend. Moreover, the interval will actually not be cleared, which means that if this component is unmounted, your interval would keep running. And the next time you come back, a second interval will be started.

If you put the `previousRole` and `currentRole` into a `useState`, you can use the `useEffect` to trigger whenever there was an update, to then trigger the setTimeout which would function as a debounce. Additionally you can also make use of the `clearTimeout` to clear a timeout when it's not relevant anymore, this could be because a new timeout has started, or because the component was unmounted.

The way you can trigger code to be exectuted before the component is actually unmounted is by doing this:
https://stackoverflow.com/questions/55020041/react-hooks-useeffect-cleanup-for-only-componentwillunmount

Reply:
ik snap niet helemaal hoe je vanuit de from dat er is gemaakt kun inhaken op de state. Ik heb voor  nu een return state aan het useEffect toegevoegd. Deze zou ervoor moeten zorgen dat de interval op juiste manier word geunmount.

Reply:
Dit is al beter, maar nog altijd niet optimaal. Nu ga je alsnog om de second kijken of de waardes veranderd zijn of niet.

Het lijkt me toch nog handig om eens te kijken naar een useEffect met dependencies:
https://react.dev/reference/react/useEffect#specifying-reactive-dependencies

Uiteindelijk kan je daardoor zoiets bekomen:

```ts
useEffect(() => {
    onSubmit({role, unit});
}, [role, unit]);
```